cmake/BOUT++functions.cmake

```diff
@@ -204,9 +204,16 @@ function(bout_add_integrated_or_mms_test BUILD_CHECK_TARGET TESTNAME)
     add_dependencies(${TESTNAME} ${BOUT_TEST_OPTIONS_EXTRA_DEPENDS})
   endif()
 
+  if (NOT BOUT_TEST_OPTIONS_PROCESSORS)
+    set(BOUT_TEST_OPTIONS_PROCESSORS 1)
+  endif()
+
   # Set the actual test command
   if (BOUT_TEST_OPTIONS_USE_RUNTEST)
-    add_test(NAME ${TESTNAME} COMMAND ./runtest ${BOUT_TEST_OPTIONS_TESTARGS})
+    add_test(NAME ${TESTNAME}
+      COMMAND ./runtest ${BOUT_TEST_OPTIONS_TESTARGS}
+      PROCESSORS ${BOUT_TEST_OPTIONS_PROCESSORS}
```

Comment:
Looks like `PROCESSORS` is not an argument to `add_test` and we need to set it as a property instead:

```cmake
  set_tests_properties(${TESTNAME} PROPERTIES
    PROCESSORS ${BOUT_TEST_OPTIONS_PROCESSORS}
    PROCESSOR_AFFINITY ON
    )
```

This should go outside this whole `if` block, L224